code/modules/mob/living/simple_animal/hostile/megafauna/_megafauna.dm

```diff
@@ -118,24 +118,45 @@
 			if(!client && ranged && ranged_cooldown <= world.time)
 				OpenFire()
 
-			if(L.health <= HEALTH_THRESHOLD_DEAD && HAS_TRAIT(L, TRAIT_NODEATH)) //Nope, it still gibs yall
+			if(L.health <= HEALTH_THRESHOLD_DEAD && HAS_TRAIT(L, TRAIT_NODEATH)) //Nope, it still kills yall
 				devour(L)
 		else
 			devour(L)
 
 /// Devours a target and restores health to the megafauna
 /mob/living/simple_animal/hostile/megafauna/proc/devour(mob/living/L)
-	if(!L)
+	if(!L || HAS_TRAIT(L, TRAIT_GUTTED))
 		return FALSE
 	visible_message(
-		span_danger("[src] devours [L]!"),
-		span_userdanger("You feast on [L], restoring your health!"))
+		span_danger("[src] devours [L]'s organs!"),
+		span_userdanger("You feast on [L]'s organs, restoring your health!"))
 	if(!is_station_level(z) || client) //NPC monsters won't heal while on station
 		adjustBruteLoss(-L.maxHealth/2)
 	L.investigate_log("has been devoured by [src].", INVESTIGATE_DEATHS)
-	L.gib()
+	var/mob/living/carbon/carbonTarget = L
+	if(istype(carbonTarget))
+		qdel(L.get_organ_slot(ORGAN_SLOT_LUNGS))
+		qdel(L.get_organ_slot(ORGAN_SLOT_HEART))
+		qdel(L.get_organ_slot(ORGAN_SLOT_LIVER))
+	L.adjustBruteLoss(500)
+	L.death() //make sure they die
+	RegisterSignal(L, COMSIG_MOB_STATCHANGE, PROC_REF(on_gutted_statchange))
+	ADD_TRAIT(L, TRAIT_GUTTED, src)
+	LoseTarget()
 	return TRUE
 
+/// When the mob's stat changes after being gutted by a megafauna (they're revived, etc)
+/mob/living/simple_animal/hostile/megafauna/proc/on_gutted_statchange(mob/living/L, new_stat)
+	SIGNAL_HANDLER
+	REMOVE_TRAIT(L, TRAIT_GUTTED, src)
+	UnregisterSignal(L, COMSIG_MOB_STATCHANGE)
```

Comment:
What if the megafauna dies by someone else's hand, then this mob is ressurected?
Will other megafauna now avoid gutting it because it has already been gutted?
I think you should change this into a status effect on the mob which applies the trait, then removes itself and the trait when the mob is revived. This makes it independent of the status of the megafauna.